flathunter/sender_slack.py

```diff
@@ -0,0 +1,54 @@
+"""Functions and classes related to sending Slack messages"""
+import json
+from typing import Dict
+
+import requests
+
+from flathunter.abstract_notifier import Notifier
+from flathunter.abstract_processor import Processor
+from flathunter.config import YamlConfig
+from flathunter.logging import logger
+
+
+class SenderSlack(Processor, Notifier):
+    """Expose processor that sends Slack messages"""
+
+    def __init__(self, config: YamlConfig) -> None:
+        self.config = config
+        self.webhook_url = self.config.slack_webhook_url()
+
+    def process_expose(self, expose: Dict) -> Dict:
+        """Send a message to a Slack channel describing the expose"""
+        message = self.config.message_format().format(
+            title=expose['title'],
+            rooms=expose['rooms'],
+            size=expose['size'],
+            price=expose['price'],
+            url=expose['url'],
+            address=expose['address'],
+            durations="" if 'durations' not in expose else expose[
+                'durations']).strip()
+        self.notify(message)
+        return expose
+
+    def notify(self, message: str) -> None:
+        """Send message to the Slack webhook"""
+        self.__send_message(message)
+
+    def __send_message(self, message: str) -> None:
+        """Send messages to the Slack webhook"""
+        logger.debug(('webhook_url:', self.webhook_url))
+        logger.debug(('message', message))
+        response = requests.post(
+            self.webhook_url,
```

Comment:
The type-checker doesn't like this line - self.webhook_url may be None:

```
/home/runner/work/flathunter/flathunter/flathunter/sender_slack.py
  /home/runner/work/flathunter/flathunter/flathunter/sender_slack.py:43:13 - error: Argument of type "Unknown | None" cannot be assigned to parameter "url" of type "str | bytes" in function "post"
    Type "Unknown | None" cannot be assigned to type "str | bytes"
      Type "None" cannot be assigned to type "str | bytes"
        Type "None" cannot be assigned to type "str"
        Type "None" cannot be assigned to type "bytes" (reportGeneralTypeIssues)
```

Reply:
I've fixed the failing test and also set the default of `self.webhook_url` to an empty string.

Reply:
Looking better. I think the nicer option would have been to abort the 'send_message' function if the webhook_url is None - it's probably also going to raise an error if you try posting data to the URL "". But I'm totally fine with this - keeps the type-checker happy and that seems like an edge case.